src/Components/Onboarding/Steps/Artists/ItemLink.tsx

```diff
@@ -49,26 +49,47 @@ interface Props extends React.HTMLProps<HTMLAnchorElement>, RelayProps {
   relay?: RelayProp
 }
 
+interface ArtistFollowProps {
+  id: string | null
+  __id: string
+}
+
 class ItemLink extends React.Component<Props, State> {
   state = {
     fadeIn: false,
     fadeOut: false,
     artist_id: this.props.artist.id,
   }
 
-  followArtist() {
-    const storeUpdater: SelectorStoreUpdater = (store, suggestedArtistData) => {
-      // Replace now followed artist in list with newly suggested artist
+  followArtist(artist: ArtistFollowProps) {
+    const storeUpdater = (store: RecordSourceSelectorProxy) => {
+      const mutationPayload = store.getRootField("followArtist")
```

Comment:
So, for the life of me, I can't figure out how to actually use the Relay store/internals to get and modify the data.

I spent a lot of time trying various combinations and organization, including refactoring.

One interesting note- a lot of examples on line show a really simple case, such as updating the name. That's actually trivial, since its a scalar: `origArtist.setValue('name', 'New Suggested Artist')` - this updates the DOM!!!

However, our thumbnail is a nested `Image` type...couldn't figure out how to use Relay/`getLinkedRecord`, etc. to retrieve or update those. It's sort of funny that the examples I found online get to essentially this point, but work with much simpler data/shapes, and so the straightforward approach works. Perhaps we're still doing something not Relay like here that is giving it trouble?

Anyways...this is what I meant about 'fighting the framework'. This would/should have been trivial without Relay, and something just feels wrong if its taking this long to figure out the proper way to do these things. I was trying to make some updates on the Metaphysics side as well: https://github.com/artsy/metaphysics/pull/802

I'm still very curious as to how to get this to work, and I'm wondering where I'm going wrong!

Reply:
There’s a rudimentary debugging tool that allows you to dump the in-memory store, which you can use like so:

```js
require("relay-runtime/lib/RelayStoreProxyDebugger").dump(store)
```

And results in the following:

<img width="1490" alt="screen shot 2017-11-13 at 15 07 20" src="https://user-images.githubusercontent.com/2320/32729553-7d31930a-c884-11e7-8c75-6989ace67be2.png">

In there I found that a record had been created as expected for the newly suggested artist and then I retrieved the ID of the suggested artist simply from the raw `data` response (the 2nd parameter to the updater function).

The resulted code changes can be found in #383.

----

The [Relay docs](https://facebook.github.io/relay/docs/relay-debugging.html#a-simple-example) mention an ‘inspector’ which seems to provide similar information, but maybe slightly more geared towards human introspection. I was able to use it like so:

```js
const { RecordSourceInspector } = require("relay-runtime")
const inspector = new RecordSourceInspector(this.props.relay.environment.getStore().getSource())
console.log(inspector.getRoot())
// use debugger to access all of the inspector API
```

----

Finally, once I’ve had the time to rebase our Relay (`__id`) patches to the latest version, we’ll be able to use the [visual debuggers](https://facebook.github.io/relay/docs/relay-debugging.html#debugging-visually).

Reply:
This is useful info! 

I ran into this problem a few times when I was experimenting in Prediction, where I wanted a `transform` function that would allow a user to intercept and modify data before being passed along to fragments.